source-azure-blob-storage/main.go

```diff
@@ -0,0 +1,212 @@
+package main
+
+import (
+	"context"
+	_ "embed"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/estuary/connectors/filesource"
+	"github.com/estuary/flow/go/parser"
+	pf "github.com/estuary/flow/go/protocols/flow"
+)
+
+//go:embed schema.json
+var schema string
+
+func handleError(err error) {
```

Comment:
We need to return errors so that they are bubbled up to the higher level handlers here: https://github.com/estuary/connectors/blob/cd7c1b12ed2aa450e24953e2d5df18143f12e4ad/source-boilerplate/boilerplate.go#L94-L96.

The whole `if err != nil { return err }` thing is one of the most horrible parts of Go, but doing it this way makes our error presentation a lot better and more consistent.